app/controllers/user.js

```diff
@@ -38,19 +40,70 @@ const emailIsRegistered = email =>
     throw errors.databaseError(e.message);
   });
 
+const createUser = (user, isAdmin = false) =>
+  bcryptService
+    .hashPassword(user.password)
+    .then(hashedPassword => User.create({ ...user, password: hashedPassword, isAdmin }));
+
 exports.signUp = ({ user }, res, next) =>
   emailIsRegistered(user.email)
     .then(isRegistered => {
       if (isRegistered) throw errors.badRequest(errorMsgs.emailIsAlreadyRegistered);
       else
-        return bcryptService.hashPassword(user.password).then(hashedPassword =>
-          User.create({ ...user, password: hashedPassword }).then(newUser => {
-            logger.info(`User ${newUser.lastName}, ${newUser.firstName} created successfuly`);
+        return createUser(user).then(newUser => {
+          logger.info(`User ${newUser.lastName}, ${newUser.firstName} created successfuly`);
+          res
+            .status(201)
+            .send(newUser)
+            .end();
+        });
+    })
+    .catch(next);
+
+const getUserForToken = token => {
+  try {
+    const decoded = jwt.decode(token);
+    return User.find({ where: { email: decoded.email } });
+  } catch (e) {
+    throw errors.badRequest(errorMsgs.invalidToken);
+  }
+};
+
+exports.createAdmin = (req, res, next) =>
+  getUserForToken(req.headers.token)
+    .then(requestingUser => {
+      if (!requestingUser) {
+        // correctly decoded token belongs to no user, perhaps it was deleted without invalidating token?
+        throw errors.internalServerError();
+      }
+
+      if (!requestingUser.isAdmin) throw errors.badRequest(errorMsgs.insufficientPermissions);
+
+      return User.find({ where: { email: req.user.email } }).then(inDbUser => {
+        if (inDbUser) {
+          // inDbUser should be updated to be admin...
+          return bcryptService.hashPassword(req.user.password).then(hashedPassword => {
```

Comment:
Why use bcryptService? You should update property admin in this user. Check update method for models.